survey/survey_app/views.py

```diff
@@ -68,3 +81,68 @@ def create(self, request, question_pk=None):
 
         return Response(serializer.data, status=status.HTTP_201_CREATED)
 
+
+class UpdateSurveyAPIView(UpdateAPIView):
+    serializer_class = SurveySerializer
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        return Survey.objects.all()
+
+    def get_object(self):
+        queryset = self.get_queryset()
+        survey = get_object_or_404(queryset, pk=self.kwargs.get('survey_id'))
+
+        # Check if the current user is the creator of the survey
+        if self.request.user != survey.creator:
```

Comment:
You should check permission in a permission class @newDaria

Reply:
view.py 
from .permissions import CanUpdateSurveyPermission
class UpdateSurveyAPIView(UpdateAPIView):
    serializer_class = SurveySerializer
    permission_classes = [IsAuthenticated, CanUpdateSurveyPermission]

    def get_queryset(self):
        return Survey.objects.all()

    def get_object(self):
        queryset = self.get_queryset()
        survey = get_object_or_404(queryset, pk=self.kwargs.get('survey_id'))
        self.check_object_permissions(self.request, survey)  # Check permissions
        return survey


permissions.py

class CanUpdateSurveyPermission(BasePermission):
    def has_permission(self, request, view):
        survey_id = view.kwargs.get('survey_id')
        try:
            survey = get_object_or_404(Survey, pk=survey_id)
        except Survey.DoesNotExist:
            # If the Survey object does not exist, return False to deny permission
            return False

        # Check if the current user is the creator of the survey
        if request.user == survey.creator:
            return True

        # Raise PermissionDenied if the user is not the creator
        raise PermissionDenied("You do not have permission to update this survey.")